src/main.js

```diff
@@ -21,19 +21,24 @@ function initialize() {
 
 		// `contextmenu` Event listener running on document.body
 		onRightClick: [],
+
+		// `keyup` Event listener running on document.body
+		onKeyUp: [],
 	};
 
 	// Run all our mods
 	const registerOnDomChange = callback => rerunning.onDomChange.push(callback);
 	const registerOnChatChange = callback => rerunning.onChatChange.push(callback);
 	const registerOnLeftClick = callback => rerunning.onLeftClick.push(callback);
 	const registerOnRightClick = callback => rerunning.onRightClick.push(callback);
+	const registerOnKeyUp = callback => rerunning.onKeyUp.push(callback);
 	mods.forEach(mod => {
 		mod.run({
 			registerOnDomChange,
 			registerOnChatChange,
 			registerOnLeftClick,
 			registerOnRightClick,
+			registerOnKeyUp,
```

Comment:
I thought about this a bit - I'd rather not expose key events to every mod.
Keypresses are pretty sensitive, and I'd prefer an individual mod not be able to track keys. Already had one person concerned the mod was keylogging (they saw part of the compiled user script code for the _keypresstracker mod and got confused).

I'm not sure exactly what system would work best, but for the time being, can you make this event listener call specifically in the mod file, rather than add it as a hook here?

Reply:
I originally had it setup the way you're describing actually (event handler attached in `screenshotMode/index.js` and the rest of the logic in a `helper.js`), but when testing found that if I switched tabs and waited for it to disconnect (`connection closed` in the console) it'd lose the handler when you switched back to the tab and the game reloaded itself. 

Switching to the way it is now manages to fix that since it re-attaches the handler if `js-uimod-initd` doesn't exist (I'm assuming that's killed during the reload).

Would there be better way to resolve that or is it just not worth fixing? It's a pretty situational problem so I could understand just letting the issue exist

Reply:
Moved handler back into mod without above issue, assuming reload losing the handler was just a previous issue on my part